src/instant.rs

```diff
@@ -67,20 +69,20 @@ impl<Clock: crate::Clock> Instant<Clock> {
     /// # fn now(&self) -> Result<Instant<Self>, embedded_time::clock::Error<Self::ImplError>> {unimplemented!()}
     /// }
     ///
-    /// assert_eq!(Instant::<Clock>::new(5).duration_since::<Microseconds<u64>>(&Instant::<Clock>::new(3)),
+    /// assert_eq!(Instant::<Clock>::new(5).duration_since::<Microseconds<u64>, _>(&Instant::<Clock>::new(3)),
     ///     Ok(Microseconds(2_000_u64)));
     ///
-    /// assert_eq!(Instant::<Clock>::new(3).duration_since::<Microseconds<u64>>(&Instant::<Clock>::new(5)),
-    ///     Err(()));
+    /// assert_eq!(Instant::<Clock>::new(3).duration_since::<Microseconds<u64>, _>(&Instant::<Clock>::new(5)),
+    ///     Err(TimeError::DisorderedOperands));
     /// ```
-    pub fn duration_since<Dur: Duration>(&self, other: &Self) -> Result<Dur, ()>
+    pub fn duration_since<Dur: Duration, E: Error>(&self, other: &Self) -> Result<Dur, TimeError<E>>
```

Comment:
It makes it necessary to add a generic to any function that might be used by a function that returns a Result<_, TimeError<E>> where E is the ImplError associated type in `Clock`.